src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystem.java

```diff
@@ -1591,6 +1595,45 @@ private void walk(IndexNode inode, Consumer<IndexNode> consumer) {
         return cen;
     }
 
+    /**
+     * Check Inode.name to see if it includes a "." or ".." in the name array
+     * @param path  the path as stored in Inode.name to verify
+     * @return true if the path contains a "." or ".." entry; false otherwise
+     */
+    private boolean hasDotOrDotDot(byte[] path) {
+        boolean dotOrDotDotFound = false;
+        // Inode.name always includes "/" in path[0]
+        assert path[0] == '/';
+        if (path.length == 1) {
+            return false;
+        }
+        int index = 1;
+        while (index < path.length) {
+            int starting = index;
+            while (index < path.length && path[index] != '/') {
+                index++;
+            }
+            // Check the path snippet for a "." or ".."
+            if (isDotOrDotDotPath(path, starting, index)) {
+                return true;
+            }
+            index++;
+        }
+        return dotOrDotDotFound;
```

Comment:
Could simply return `false` here, and eliminate the local variable `dotOrDotDotFound`.